### DISCUSSION TITLE
Missing validation loss in Object detection

### DISCUSSION THREAD

**Discussion by AashishKumbashi** [azavea/raster-vision#1957] [Q&A] [answered]
I'm currently working on object detection using Raster Vision as a library. While I'm new to this, everything seems to be functioning well. However, I'm interested in understanding how to incorporate validation loss into the object detection process.

The metrics I currently have access to are displayed in the image below:
![Screenshot from 2023-10-12 15-23-12](https://github.com/azavea/raster-vision/assets/104180642/cfe6b310-325c-4971-95ec-95a71f0e16ea)

Any guidance on how to implement validation for the object detection task using Raster Vision would be greatly appreciated.

**Comment by AdeelH** [comment, marked as answer]
TorchVision's implementations of FasterRCNN and other detection models unfortunately do not return losses during inference--just the predictions. See, e.g., https://github.com/pytorch/vision/issues/1350.

**Reply by AashishKumbashi** [reply]
Thanks for pointing me to this. Do you know of any other way in which I can determine overfitting using this?

**Reply by AdeelH** [reply]
>Do you know of any other way in which I can determine overfitting using this?

I guess you could track map/map50.

**Reply by AashishKumbashi** [reply]
I am still not sure how to track overfitting using map and map50. Tried looking for resources online but to no avail. It would be great if you could provide me with more details or point me to a resource that would help me track overfitting using Rastervision for object detection.

**Reply by AdeelH** [reply]
I meant that you could think of map or map50 as any other metric and consider the model to have overfit if its value starts going down.

There is now also a [`save_all_checkpoints` option in `LearnerConfig`](https://docs.rastervision.io/en/stable/api_reference/_generated/rastervision.pytorch_learner.learner_config.LearnerConfig.html#rastervision.pytorch_learner.learner_config.LearnerConfig.save_all_checkpoints). So you could just pick the checkpoint where the map was highest.
